docs/creating_service_module.md

```diff
@@ -0,0 +1,30 @@
+## Creating a new Service Module project in this repository
```

Comment:
@secondsun @paolobueno I'm confused by the terminology. By _project_ here do we mean a _module_ or an entire new project consisting of many modules? As far as I understand, the repository `aerogear-android-sdk` is the project and inside there are many modules (i. e. auth, core).

I think it is a bit mixed throughout the many docs. Moreover there is also mention of "subproject" which adds a bit more of inconsistency.